### ISSUE TITLE
NoSuchMethodError on setup task

### ISSUE THREAD

**Issue by kashike** [MinecraftForge/ForgeGradle#247]
https://gist.github.com/kashike/8a4ca01e1778fda7f41c

**Comment by AbrarSyed** [comment]
Unable to reproduce... anyone else?

**Comment by LexManos** [comment]
What wrapper version?

**Comment by kashike** [comment]
This seems to only happen on 2.6.

2.4 installed: success
2.4 wrapper:
2.5 installed: success
2.5 wrapper: success
2.6 installed: fail
2.6 wrapper: fail

**Comment by AbrarSyed** [comment]
Your 2.6 wrapper be broke man.. That's all there is to it. Redownload or something.

**Comment by kashike** [comment]
I'm using the latest release of Gradle, and I did re-download it already.

**Comment by AbrarSyed** [comment]
In that case.. I am truly stumped. Perhaps try gradle support? Its really weird that only you have this issue...

**Comment by Lunatrius** [comment]
I can reproduce this. Gradle 2.6 install fails with the same stack trace. Could it be that it's trying to find the method in `BuildLauncher` instead of the `ConfigurableLauncher` parent interface?

**Comment by AbrarSyed** [comment]
unlikely.. its not a static method or anything.. it should compile to a virtual call that decides inheritance at runtime...

**Comment by Lunatrius** [comment]
The resulting binaries actually differ.

Official FG2 jar:
```java
connection.newBuild().setStandardOutput(System.out).setStandardInput(System.in).setStandardError(System.err).withArguments((String[])args.toArray(new String[args.size()])).setColorOutput(false).run();
```
Compiled using Gradle 2.6:
```java
((BuildLauncher)((BuildLauncher)((BuildLauncher)((BuildLauncher)((BuildLauncher)connection.newBuild().setStandardOutput(System.out)).setStandardInput(System.in)).setStandardError(System.err)).withArguments((String[])args.toArray(new String[args.size()]))).setColorOutput(false)).run();
```
Could be most likely solved by not chaining the calls. A lot of the methods have been pulled up into `ConfigurableLauncher<T>` ([2.6](https://docs.gradle.org/2.6/javadoc/org/gradle/tooling/BuildLauncher.html#method_summary) vs [2.5](https://docs.gradle.org/2.5/javadoc/org/gradle/tooling/BuildLauncher.html#method_summary) docs).
```java
BuildLauncher buildLauncher = connection.newBuild();
buildLauncher.setStandardOutput(System.out)
buildLauncher.setStandardInput(System.in)
buildLauncher.setStandardError(System.err)
buildLauncher.withArguments(args.toArray(new String[args.size()]))
buildLauncher.setColorOutput(false)
buildLauncher.run();
```

**Comment by cpw** [comment]
That looks like a pretty bad gradle regression to be honest..

**Comment by AbrarSyed** [comment]
Can someone try this with the Gradle 2.7 RC? just curious to see if they fixed it or now in the new release...

**Comment by AbrarSyed** [comment]
AHAH!

"IMPORTANT A performance regression was introduced in the Gradle wrapper in Gradle 2.6. If you generated your wrapper with 2.6, we recommend that you regenerate your wrapper with 2.7. If you generated your wrapper with a older version, you can still configure that wrapper to point to Gradle 2.6 without issue."

So its fixed in Gradle 2.7..
